app/views.py

```diff
@@ -207,6 +275,7 @@ def createquizsuccess(quizID):
 #    return render_template('sidebar.html', image_file=image_file, title=' | Class')
 
 @app.route('/class/<int:groupID>/code')
+@login_required
 def update_class_code(groupID):
     """Routing to update Class Code"""
     group = validate_group_link(groupID)
```

Comment:
I realized there is no backend validation for this and some of the functions to check if it is an educator? What if a student types in the routing address?

Reply:
This was done by you. Will have to update then.

Reply:
It has validation I realized -> line 213, 214. Just help change the error there from 404 to 403